File: labs/python/hello/test_main.py
import unittest

from main import greet


class GreetTest(unittest.TestCase):
    def test_default(self):
        self.assertEqual(greet(), "Hello, world!")

    def test_name(self):
        self.assertEqual(greet("Claude"), "Hello, Claude!")


if __name__ == "__main__":
    unittest.main()

File: labs/cpp/hello/test_main.cpp
// Minimal assertion-based tests: no framework, non-zero exit on failure.
#include <iostream>

#include "greet.hpp"

static int failures = 0;

#define CHECK_EQ(actual, expected)                                           \
    do {                                                                     \
        if ((actual) != (expected)) {                                        \
            std::cerr << __FILE__ << ":" << __LINE__ << ": " #actual         \
                      << " == \"" << (actual) << "\", expected \""           \
                      << (expected) << "\"\n";                               \
            ++failures;                                                      \
        }                                                                    \
    } while (0)

int main() {
    CHECK_EQ(greet(), "Hello, world!");
    CHECK_EQ(greet("Claude"), "Hello, Claude!");

    if (failures == 0) std::cout << "all tests passed\n";
    return failures == 0 ? 0 : 1;
}
